src/web/web_views.nit

```diff
@@ -119,3 +119,41 @@ class HtmlDocPage
 		return tpl
 	end
 end
+
+# Display the source for each mentities
+class HtmlDotPage
+	super NitView
+
+	# Where to output dot images.
+	var out_file: String
+
+	# Dot to process.
+	var dot: Text
+
+	# Page title.
+	var title: String
+
+	redef fun render(srv) do
+		var tpl = new Template
+		tpl.add new Header(1, title)
+		tpl.add render_dot
+		return tpl
+	end
+
+	private fun render_dot: String do
+		var path = out_file
+		var file = new FileWriter.open("{path}.dot")
+		file.write(dot)
+		file.close
+		var proc = new ProcessReader("dot", "-Tsvg", "-Tcmapx", "{path}.dot")
```

Comment:
It is funny, you choose to pass your input as a file, but read the output as a stream.